blog/2024-08-08-quroum-queues-in-4.0/index.md

```diff
@@ -0,0 +1,197 @@
+and use the priorities just to make sure that in case of an occasional long backlog, important messages
+can skip the queue.
+1. If you really need more priorities and/or more control over how different priorities are handled,
+using multiple queues is your best bet. You can develop a consumer that subscribes to multiple queues
+and then decides which queue to consume from.
+
+## Consumer Priorities with Single Active Consumer
+
+Starting with RabbitMQ 4.0, a quorum queue can switchover from one single active consumer
+to another, higher priority, consumer. This is particularly useful if you have multiple queues
+that should have a single consumer each, but you don't want a single instance of your application
+to be the consumer for all of them, which is likely to happen when the first application instance that starts,
+subscribes to all those single-active-consumer queues. Now you can pick a different priority when subscribing
+to different queues, to make sure that each instance consumes only from its "favorite" queue and only serves
+as a backup consumer for other queues.
+
+
+To better explain this functionality, let's review all the moving parts.
+A [Single Active Consumer](/docs/consumers#single-active-consumer) is a queue argument which
+prevents a queue from delivering messages to more than one consumer, regardless of how many
+subscribed to the queue. One consumer is active, all other consumers are not. If the active
+consumer disconnects, one of the other consumers is activated. This feature is used
+if a strict message processing order needs to be maintained.
+
+[Consumer Priority](/docs/consumers#priority) allows you to specify that rather than delivering
+messages to all subscribed consumers in a fair round-robin fashion (which is the default behavior
+of both classic and quorum queues), a certain consumer should be preferred.
+
+Until version 4.0, these features were effectively mutually exclusive - if Single Active Consumer was enabled,
+a new consumer would never become active, regardless of its priority, as long as the previous consumer remained active.
+Starting with 4.0, if the new consumer's priority is higher than the currently active consumer's, the quorum queue
+will switchover to the higher priority consumer: it will stop delivering messages to the current
+consumer, wait for all the messages to be acknowledged, and then will deactivate the old consumer,
+and activate the higher priority consumer instead.
+
+The graph below shows this behavior. There are three metrics on this graph:
+* the green line shows the number of messages consumed by the first (default priority) consumer (which happens to be configured to consume 10 msgs/s)
+* yellow, shows the same value but for the second, higher priority consumer
+* blue, shows the number of unacknowledged messages (axis scale on the right)
+
+![Single Active Consumer Switchover](sac-and-consumer-priority.png)
```

Comment:
"Single Active Consumer Switchover: The Normal-Priority Consumer (Green) Gets Deactivated When It Has Acknowledged Its Messages, Then The Higher-Priority Consumer (Yellow) Gets Activated"?